src/components/network-chart/index.js

```diff
@@ -0,0 +1,235 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import * as React from 'react';
+import explicitConnect from '../../utils/connect';
+import VirtualList from '../shared/VirtualList';
+import NetworkChartEmptyReasons from './NetworkChartEmptyReasons';
+import MarkerTooltipContents from '../shared/MarkerTooltipContents';
+import Tooltip from '../shared/Tooltip';
+
+import {
+  selectedThreadSelectors,
+  getCommittedRange,
+  getProfileInterval,
+  // getPreviewSelection,
+} from '../../reducers/profile-view';
+import { getSelectedThreadIndex } from '../../reducers/url-state';
+import { updatePreviewSelection } from '../../actions/profile-view';
+
+import type { TracingMarker } from '../../types/profile-derived';
+import type {
+  Milliseconds,
+  UnitIntervalOfProfileRange,
+} from '../../types/units';
+import type {
+  ExplicitConnectOptions,
+  ConnectedProps,
+} from '../../utils/connect';
+
+require('./index.css');
+
+const ROW_HEIGHT = 16;
+
+type DispatchProps = {|
+  +updatePreviewSelection: typeof updatePreviewSelection,
+|};
+
+type StateProps = {|
+  +markers: TracingMarker[],
+  +timeRange: { start: Milliseconds, end: Milliseconds },
+  +interval: Milliseconds,
+  +threadIndex: number,
+|};
+
+type Props = ConnectedProps<{||}, StateProps, DispatchProps>;
+
+class NetworkChartRow extends React.PureComponent<Props> {
+  constructor(props: any) {
+    super(props);
+    this.state = { pageX: null, pageY: null, hovered: false };
+    (this: any)._hoverIn = this._hoverIn.bind(this);
+    (this: any)._hoverOut = this._hoverOut.bind(this);
+    (this: any)._shortenURI = this._shortenURI.bind(this);
+  }
+
+  _hoverIn(event) {
```

Comment:
please type the parameter explicitely; in this case this is a `SyntheticMouseEvent<>` (Note we don't need to type the generic part because that's what types the `currentTarget` property but you don't use it here).